### PR TITLE
Implemented Idint

### PR DESCRIPTION
**Pull request by parth121101**
Continuation of #3177

### CODE REVIEW

**Review by Pranavchiku — APPROVED** [review]
Addresed https://github.com/lfortran/lfortran/pull/3177#discussion_r1465071048, rest all looks good to me. Thanks!

File: integration_tests/intrinsics_128.f90 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+program intrinsics_128
+    double precision :: x
+    x = 4.23
```
Comment (certik) [inline integration_tests/intrinsics_128.f90:3]: You have to write this as:
`4.23d0`
otherwise it's a single precision number, not double precision.

Comment (Pranavchiku) [reply integration_tests/intrinsics_128.f90:3]: Ahh okay, yes. I thought of this, but then LFortran casts it in ASR, hence I enabled auto-merge. I am sorry for it, will take care from next time.

Comment (certik) [reply integration_tests/intrinsics_128.f90:3]: Yes, it gets cast, but as a Fortran programmer, if you want to get `4.230000000000000` and not `4.23000000134583745`, you need to use `d0` or `_dp`.

Comment (Pranavchiku) [reply integration_tests/intrinsics_128.f90:3]: Okay, yes. I will take care :)

**Pull request merged**